projects/packages/account-protection/src/class-validation-service.php

```diff
@@ -253,14 +296,23 @@ public function is_current_password( int $user_id, string $password ): bool {
 	/**
 	 * Check if the password has been used recently by the user.
 	 *
-	 * @param int    $user_id The user ID.
-	 * @param string $password The password to check.
+	 * @param \WP_User|\stdClass $user The user data.
+	 * @param string             $password The password to check.
 	 *
-	 * @return bool True if the password hash was recently used, false otherwise.
+	 * @return bool True if the password was recently used, false otherwise.
 	 */
-	public function is_recent_password( int $user_id, string $password ): bool {
-		$recent_passwords = get_user_meta( $user_id, Config::VALIDATION_SERVICE_RECENT_PASSWORD_HASHES_USER_META_KEY, true );
+	public function is_recent_password( $user, string $password ): bool {
+		// Skip on create-user validation - no user ID yet
+		if ( empty( $user->ID ) ) {
+			return false;
+		}
```

Comment:
Why does this `is_recent_password` method need to support a case where the user has no ID?

IMO It should be up to the implementation to conditionally call a method like this.

Reply:
In the `user-new.php` form the user object passed does not have a `user->ID`  since the user hasn't been created yet and this errors out. Because we are only running the validation on `user_profile_update_errors` or `validate_password_reset` there didn't appear to be a simple way to conditionally allow this to run in the profile update but during in user creation. This seemed to be a reasonable way to bypass the need for extra logic, since were always aware that this will return `false` in that case.

Reply:
Nvm, I stand corrected - I appear to have since found a way to circumnavigate the need for that check but neglected to return to remove it.